doc: add operator-scope kb layout (#2982)
operator-framework/operator-sdk: website/content/en/docs/operator-scope.md

weight: 50
---

- [Namespace-scoped operator usage](#namespace-scoped-operator-usage)
- [Cluster-scoped operator usage](#cluster-scoped-operator-usage)
  - [Changed required for a cluster-scoped operator](#changed-required-for-a-cluster-scoped-operator)
  - [Example for cluster-scoped operator](#example-for-cluster-scoped-operator)
- [CRD scope](#crd-scope)
  - [CRD cluster-scoped usage](#crd-cluster-scoped-usage)
  - [Example for changing the CRD scope from namespace to cluster](#example-for-changing-the-crd-scope-from-namespace-to-cluster)

> 

Author: Remove the menu becuase:
1 - It would no longer have the CRD scope
2- The docs has no menu so we should keep the same standard across the project

weight: 50
---

## Overview

A namespace-scoped operator watches and manages resources in a single namespace, whereas a cluster-scoped operator watches and manages resources cluster-wide. Namespace-scoped operators are preferred because of their flexibility. They enable decoupled upgrades, namespace isolation for failures and monitoring, and differing API definitions.

However, there are use cases where a cluster-scoped operator may make sense. For example, the [cert-manager](https://github.com/jetstack/cert-manager) operator is often deployed with cluster-scoped permissions and watches so that it can manage issuing certificates for an entire cluster.

**NOTE**: CustomResourceDefinition (CRD) scope can also be changed to cluster-scoped. See the [CRD scope][crd-scope-doc] document for more details.

## Namespace-scoped operator usage

This scope is ideal for operator projects which will control resources just in one namespace, which is where the operator is deployed.

> **NOTE:** Initial projects created by `operator-sdk` are namespace-scoped by default which means that it will NOT have a `ClusterRole` defined in the `deploy/role_binding.yaml`.

Reviewer: Even though this line wasn't added here, we should still correct the grammar a bit
```suggestion
**NOTE:** Projects created by `operator-sdk` are namespace-scoped by default which means that they will NOT have a `ClusterRole` defined in `deploy/`.
```

Author: not part of the scope of this PR but ok we can address either.

However, there are use cases where a cluster-scoped operator may make sense. For example, the [cert-manager](https://github.com/jetstack/cert-manager) operator is often deployed with cluster-scoped permissions and watches so that it can manage issuing certificates for an entire cluster.

> **NOTE**:CustomResourceDefinition (CRD) scope can also be changed for cluster-scoped, for further information see the [CRD Scope][crd-scope-doc] document.

Author: Note that it is the legacy doc and it is not part of the scope. 
However, has no reason for do not fix. So, done 👍

> **NOTE**:CustomResourceDefinition (CRD) scope can also be changed for cluster-scoped, for further information see the [CRD Scope][crd-scope-doc] document.

Author: Done. 👍

      name: memcached-operator
      namespace: <operator-namespace>
    ```

> 

Author: Just moved to be ref.

Reviewer 2: Where is the reference to that new doc?
Shouldn't we be linking to it from here?
